### PR TITLE
fix: support azure speech running in containers

### PR DESCRIPTION
**Pull request by longcw**
Add support for azure speech containers: https://learn.microsoft.com/en-us/azure/ai-services/speech-service/speech-container-stt?tabs=container&pivots=programming-language-python#use-the-container

### CODE REVIEW

File: livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py [hunk 1]
```diff
@@ -135,13 +138,14 @@ def __init__(
...
+        speech_host = speech_host or os.environ.get("AZURE_SPEECH_HOST")
         speech_key = speech_key or os.environ.get("AZURE_SPEECH_KEY")
-        if not speech_key:
-            raise ValueError("AZURE_SPEECH_KEY must be set")
-
         speech_region = speech_region or os.environ.get("AZURE_SPEECH_REGION")
-        if not speech_region:
-            raise ValueError("AZURE_SPEECH_REGION must be set")
+
+        if not speech_host and (not speech_key or not speech_region):
```
Comment (davidzhao) [inline livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py:145]: nit: slightly more readable
 
```suggestion
        if not speech_host and not (speech_key and speech_region):
```

**Review by davidzhao — APPROVED** [review]
lg, just one nit

File: livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py [hunk 2]
```diff
@@ -271,11 +275,14 @@ def close(self) -> None:
...
-        speech_recognition_language=config.language or "en-US",
-    )
+    if config.speech_host:
+        speech_config = speechsdk.SpeechConfig(host=config.speech_host)
+    else:
+        speech_config = speechsdk.SpeechConfig(
+            subscription=config.speech_key,
+            region=config.speech_region,
+            speech_recognition_language=config.language or "en-US",
+        )
```
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py:285]: When there is a custom host, we don't need to configure the other fields?

**Review by theomonnom — APPROVED** [review]

File: livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py (hunk 2 above)
Comment (longcw) [reply livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py:285]: Yes, it only needs `speech_config = speechsdk.SpeechConfig(
    host="http://localhost:5000")` (ref https://learn.microsoft.com/en-us/azure/ai-services/speech-service/speech-container-ntts?tabs=container&pivots=programming-language-python#use-the-container)

**Pull request merged**